### DISCUSSION TITLE
Varying results RT coverage map

### DISCUSSION THREAD

**Discussion by theTobi01** [NVlabs/sionna#447] [Q&A]
Hi everyone,

I am working with Sionna RT (currently with version 0.16, I didn't update yet) using the coverage map function.

While trying out different resolutions for the coverage map, I noticed that the results vary a lot depending on the resolution.
Here an example (I am working in a custom 3D model):

`cm = scene.coverage_map(cm_cell_size=[5, 5], num_samples=int(10e6), max_depth = 6)`
`cm.show()`
![cm_res_5](https://github.com/NVlabs/sionna/assets/92628687/44ee1f5f-46fe-4d95-9c38-257dfdbd5040)

`cm = scene.coverage_map(cm_cell_size=[1, 1], num_samples=int(10e6), max_depth = 6) `
`cm.show()`
![cm_res_1](https://github.com/NVlabs/sionna/assets/92628687/7cacd9de-a8c0-45c2-935d-1086deb68aad)

`cm = scene.coverage_map(cm_cell_size=[0.1, 0.1], num_samples=int(10e6), max_depth = 6)`
`cm.show()`
![cm_res_0 1](https://github.com/NVlabs/sionna/assets/92628687/abb5e3f1-e8b1-4b51-8ac7-bf8d9f7c9735)

As you can see here, the results for lower resolutions (1m and 5m) are similar, which makes sense to me. But the higher resolution (0.1m) seams "cut off". The coverage is way worse than with the other ones.

I didn't really go through the code for the cm.solver yet, but to me it looks like an issue with very small values getting cut off. For the higher resolution the power values per element decrease of course. Is this maybe a numerical issue of some sort? And is there something I can do to prevent it and create coverage maps with a high resolution?

Thank You and Best Regards
Tobi

**Comment by jhoydis** [comment]
Hi @theTobi01,

In general, the higher the resolution, the more rays you need to shoot to make sure that the cells are actually hit by a ray.
The path loss value per cell is independent of the resolution.

However, in your case, there seems indeed something to be strange for the last figure. 

Could you share a code snippet that reproduces this behavior for one of the scenes already included with Sionna?

**Comment by theTobi01** [comment]
Hi @jhoydis,

thanks for the fast answer!

The same results appear when trying it with the etoile scene you provide in sionna:

my code:

```py
scene = load_scene(sionna.rt.scene.etoile)

scene.tx_array = PlanarArray(num_rows = 1, num_cols = 1,
                             vertical_spacing=0.5, horizontal_spacing=0.5,
                             pattern="dipole", polarization="V")
scene.rx_array = scene.tx_array

scene.frequency = 2.4e9 # 2.4 GHz

position0 = [0.0, 0.0, 0.0]
tx0 = Transmitter("tx0", position=position0)
scene.add(tx0)

cm = scene.coverage_map(cm_cell_size=[0.1, 0.1], num_samples=int(10e6), max_depth = 6)

fig = cm.show()
fig.savefig("bilder/cm_etoile_0_1.png")
![cm_etoile_0_1](https://github.com/NVlabs/sionna/assets/92628687/6ce55dbe-9b3b-4c60-af7c-9e5aca549fe6)

cm = scene.coverage_map(cm_cell_size=[1, 1], num_samples=int(10e6), max_depth = 6)

fig = cm.show()
fig.savefig("bilder/cm_etoile_1.png")
```

![cm_etoile_1](https://github.com/NVlabs/sionna/assets/92628687/34fcda31-a28c-457d-91ad-6cf131e7dc48)
